hibernate-core/src/main/java/org/hibernate/query/sqm/produce/function/ArgumentTypesValidator.java

```diff
@@ -128,43 +141,63 @@ public void validate(
 		}
 	}
 
-	private JdbcType getJdbcType(
-			MappingMetamodel metamodel,
+	private int getJdbcType(
+			TypeConfiguration typeConfiguration,
 			SqmTypedNode<?> argument,
 			JdbcTypeIndicators indicators,
 			JavaType<?> javaType) {
 		// For enum types, we must try to resolve the JdbcMapping of a possible path
 		// to be sure we use the correct JdbcType for the validation
-		final JdbcMapping mapping = javaType.getJavaTypeClass().isEnum()
-				? getJdbcMapping( argument, metamodel )
-				: null;
-		if ( mapping == null ) {
-			return javaType.getRecommendedJdbcType( indicators );
+		final JdbcType jdbcType;
+		if ( javaType.getJavaTypeClass().isEnum() ) {
+			// we can't tell from the enum class whether it is mapped
+			// as STRING or ORDINAL, we have to look at the entity
+			// attribute it belongs to when it occurs as a path expression
+			final JdbcMapping mapping = getJdbcMappingForEnum( argument, typeConfiguration );
+			if ( mapping == null ) {
+				jdbcType = javaType.getRecommendedJdbcType( indicators );
+			}
+			else {
+				// indicates that we don't know if its STRING or ORDINAL
+				return ENUM_UNKNOWN_JDBC_TYPE;
+			}
 		}
 		else {
-			return mapping.getJdbcType();
+			jdbcType = javaType.getRecommendedJdbcType( indicators );
 		}
+		return jdbcType.getDefaultSqlTypeCode();
 	}
 
-	private JdbcMapping getJdbcMapping(SqmTypedNode<?> argument, MappingMetamodel metamodel) {
+	private JdbcMapping getJdbcMappingForEnum(SqmTypedNode<?> argument, TypeConfiguration typeConfiguration) {
 		if ( argument instanceof SqmPath<?> ) {
 			final SqmPath<?> path = (SqmPath<?>) argument;
-			final ModelPartContainer modelPartContainer = getModelPartContainer( path.getLhs(), metamodel );
+			final ModelPartContainer modelPartContainer = getModelPartContainer( path.getLhs(), typeConfiguration );
 			final ModelPart part = modelPartContainer.findSubPart( path.getReferencedPathSource().getPathName(), null );
```

Comment:
I may be wrong but probably we have have to check that `modelPartContainer` is not null before calling `findSubPart`

Reply:
@dreab8 in the end that code got removed in a later commit.

Reply:
sorry for the noise